Fix column naming strategy for embedded fields (for next major version)

radovanradic opened the pull request:
This [issue](https://github.com/micronaut-projects/micronaut-data/issues/2439) is reported while ago and the fix is also needed for recent [issue](https://github.com/micronaut-projects/micronaut-data/issues/3407)

radovanradic committed:
Fix schema generator and query execution when entities have dynamic/configurable mapping value (for Oracle DB) (#3393)

* Attempt to implement property replacement for Oracle when uppercased

* Fix quoting logic

* Fix cascade operation

* Fixed resolving of query placeholder values

* Trigger build for branch temporary

* Remove debug

* Revert "Trigger build for branch temporary"

This reverts commit c9fdf30d9ff59ea76b2d1f1d2563a9ebb6fd58a9.

renovate[bot] committed:
fix(deps): update dependency io.micronaut:micronaut-core-bom to v4.8.12 (#3400)

* fix(deps): update dependency io.micronaut:micronaut-core-bom to v4.8.12

* Use mysql:8.4.5 image

* Use mysql:8.4.5 image

* Use mysql:8.4.5 image

---------

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: radovanradic <radicr@gmail.com>

radovanradic committed:
Fix for EmbeddedId regression in 4.0

radovanradic committed:
Merge branch '4.12.x' into embeddedid-colname

renovate[bot] committed:
chore(deps): update softprops/action-gh-release action to v2.2.2 (#3404)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency org.springframework.boot:spring-boot-gradle-plugin to v3.4.5 (#3408)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update spring data (#3406)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

radovanradic committed:
Don't use address prefix for embedded field

radovanradic committed:
Merge branch '4.12.x' into embeddedid-colname

radovanradic committed:
Add one more example for embedded and composite id

radovanradic committed:
Disable gradle caching temporary

radovanradic committed:
Revert gradle caching

radovanradic committed:
Add more tests for repo

radovanradic committed:
Remove trigger builds from workflow

radovanradic commented on data-processor/src/test/groovy/io/micronaut/data/model/query/builder/SqlQueryBuilderSpec.groovy:
```diff
@@ -169,7 +169,7 @@ interface MyRepository {
         def encoded = encoder.buildQuery(AnnotationMetadata.EMPTY_METADATA, q)
 
         expect:
-        encoded.query.startsWith('SELECT restaurant_.`id`,restaurant_.`name`,restaurant_.`address_street`,restaurant_.`address_zip_code`,restaurant_.`hqaddress_street`,restaurant_.`hqaddress_zip_code` FROM')
```
I think when there is 
```
@Relation(Relation.Kind.EMBEDDED)
private final Address address;
```
or 
```
@EmbeddedId
private PrimaryKey primaryKey;
```
then column should not be prefixed with `address_` or `primary_key_` and this was issue since while ago. This PR should fix it. Can cause some regression issue for existing apps, but the workaround would be adding `@MappedProperty` on embedded entity field.

graemerocher replied:
I don't think this is correct, and we can't cause regressions otherwise it has to go into next major version

radovanradic replied:
Yes, we can postpone it for 5.0

radovanradic commented on data-processor/src/test/groovy/io/micronaut/data/processor/sql/ColumnTransformerSpec.groovy:
```diff
@@ -152,6 +152,6 @@ class Project {
             def sql = builder.buildQuery(AnnotationMetadata.EMPTY_METADATA, QueryModel.from(entity).eq("xyz", new QueryParameter("xyz"))).query
 
         expect:
-            sql == 'SELECT transform_."project_id_department_id",transform_."project_id_project_id",UPPER(xyz@abc) AS xyz FROM "transform" transform_ WHERE (transform_."xyz" = LOWER(?))'
```
One example where this looked wrong, should not have `project_id_` prefix for composite embedded id.

dstepanov commented:
IMHO the prefixes are fine to me, if somebody doesn't want them, `MappedProperty` on an field in the embedded class should override the default. Can you check how does Hibernate generated the defaults?

radovanradic commented:
> IMHO the prefixes are fine to me, if somebody doesn't want them, `MappedProperty` on an field in the embedded class should override the default. Can you check how does Hibernate generated the defaults?

In Hibernate, this embeddable entity
```
@Embeddable
public record Address(

    String street,
    String zipCode) {
}
```
and actual entity using it
```
@Entity
public record EmbeddedEntity(
    @Id
    Long id,
    @Embedded
    Address address) {
}
```
generate this create SQL 
```
CREATE TABLE `embedded_entity` (
  `id` bigint NOT NULL,
  `street` varchar(255) DEFAULT NULL,
  `zip_code` varchar(255) DEFAULT NULL,
  PRIMARY KEY (`id`)
)
```

radovanradic committed:
Fix issue with CursoredPageable and @EmbeddedId (#3411)

* Use all embedded properties to sort cursored pageable

* Use PersistentPropertyPath for cursored pageable

* Trigger build to see if changes will pass tests

* Add test for the changes and remove trigger build

* Sonar fixes

mikehearn committed:
Improve the Oracle client info feature. (#3405)

1. Fix an exception that could occur if class/module names were too long.
2. Propagate the Java thread name in CLIENT_INFO if none is set from annotation metadata.
3. Add a useful utility query to the documentation.

renovate[bot] committed:
fix(deps): update dependency io.micronaut:micronaut-core-bom to v4.8.13 (#3413)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency io.micronaut.coherence:micronaut-coherence-bom to v5.0.6 (#3414)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

micronaut-build committed:
[skip ci] Release v4.12.1

micronaut-build committed:
chore: Bump version to 4.12.2-SNAPSHOT

radovanradic committed:
Merge branch '4.12.x' into embeddedid-colname

radovanradic committed:
Merge remote-tracking branch 'origin/5.0.x' into embeddedid-colname

radovanradic committed:
Merge remote-tracking branch 'origin/5.0.x' into embeddedid-colname

radovanradic committed:
Fix tests

radovanradic committed:
Revert formatting changes.

radovanradic committed:
Example for EmbeddedId naming strategy change

radovanradic committed:
Keep backward compatible EmbeddedId naming

radovanradic committed:
Add breaking changes documentation about composite id naming (@EmbeddedId)

radovanradic committed:
Updated breaking changes documentation and tests.

radovanradic committed:
Remove duplicated test example

radovanradic committed:
Merge remote-tracking branch 'origin/5.0.x' into embeddedid-colname

radovanradic committed:
Merge remote-tracking branch 'origin/5.0.x' into embeddedid-colname

radovanradic committed:
Merge remote-tracking branch 'origin/5.0.x' into embeddedid-colname

radovanradic committed:
Merge remote-tracking branch 'origin/5.0.x' into embeddedid-colname

radovanradic committed:
Fix jakarta data test in data-processor

radovanradic committed:
Merge remote-tracking branch 'origin/5.0.x' into embeddedid-colname

radovanradic committed:
Merge remote-tracking branch 'origin/5.0.x' into embeddedid-colname

radovanradic committed:
Merge remote-tracking branch 'origin/5.0.x' into embeddedid-colname

radovanradic committed:
Merge remote-tracking branch 'origin/5.0.x' into embeddedid-colname

radovanradic committed:
Merge remote-tracking branch 'origin/5.0.x' into embeddedid-colname

radovanradic committed:
Merge remote-tracking branch 'origin/5.0.x' into embeddedid-colname

radovanradic committed:
Merge remote-tracking branch 'origin/5.0.x' into embeddedid-colname

radovanradic committed:
Fix test after main branch merge
